Paws4Me/Paws4Me/Display And Save Pets/View Controllers/LocalPetNameViewController.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  LocalPetNameViewController.swift
+//  Paws4Me
+//
+//  Created by Kirsty-Lee Walker on 2022/03/10.
+//
+
+import CoreData
+import UIKit
+
+class LocalPetViewController: UIViewController, UITableViewDelegate {
+
+    // MARK: - IBOutlets
+    @IBOutlet weak private var petNameTableView: UITableView!
+    private let context = (UIApplication.shared.delegate as? AppDelegate)?.persistentContainer.viewContext
+
+    // MARK: - Vars/Lets
+    private var namePet = ""
+    private var imagePet = ""
+    private var pets: [Pet]? = []
+    private lazy var petLocalDatabaseViewModel = PetLocaldatabaseViewModel(repository: SavedPetDataRepository(),
+                                                                           delegate: self)
+
+    // MARK: - Life cycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTableView()
+        petLocalDatabaseViewModel.fetchPetDataResults()
+        petLocalDatabaseViewModel.savePetInLocalDatabase(name: namePet, image: imagePet)
+    }
+
+    // MARK: - Functions
+    func setupTableView() {
+        petNameTableView.dataSource = self
+        petNameTableView.delegate = self
+        title = "Saved Pets"
+    }
+
+    func setNamePet(name: String) {
+        self.namePet = name
+    }
+
+    func setImagePet(image: String) {
+        self.imagePet = image
+    }
+}
+
+// MARK: - UITableViewDataSource
+extension LocalPetViewController: UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return petLocalDatabaseViewModel.petSavedCount
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let pet =  petLocalDatabaseViewModel.savedPet(at: indexPath.row)
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "favouritePetCell",
+                                                       for: indexPath) as? FavouriteTableViewCell
+        else { return UITableViewCell() }
+
+        guard let namePet = petLocalDatabaseViewModel.petName(pet: pet) else { return UITableViewCell() }
+        guard let imagePet = petLocalDatabaseViewModel.petImage(pet: pet) else { return UITableViewCell() }
+        cell.updateUI(namePet: namePet, imagePet: imagePet)
+        cell.setNeedsLayout()
+        return cell
+    }
+
+    // MARK: - Delete saved pet
+    func tableView(_ tableView: UITableView,
+                   trailingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+        let deleteAction = UIContextualAction(style: .normal,
+                                              title: "Delete",
+                                              handler: { (_: UIContextualAction, _: UIView, success: (Bool) -> Void) in
+            DispatchQueue.main.async {
```

Comment:
You should move the "DispatchQueue.main.async" to your Save pet data repository

Reply:
Will do so 👍🏻